pkg/virt-controller/watch/migration.go

```diff
@@ -791,6 +797,56 @@ func (c *MigrationController) createAttachmentPod(migration *virtv1.VirtualMachi
 	return nil
 }
 
+func (c *MigrationController) handlePendingPodTimeout(migration *virtv1.VirtualMachineInstanceMigration, vmi *virtv1.VirtualMachineInstance, pod *k8sv1.Pod) error {
+	if pod.Status.Phase != k8sv1.PodPending || pod.DeletionTimestamp != nil {
+		return nil
+	}
+
+	isUnschedulable := false
+	for _, condition := range pod.Status.Conditions {
+		if condition.Type == k8sv1.PodScheduled &&
+			condition.Status == k8sv1.ConditionFalse &&
+			condition.Reason == k8sv1.PodReasonUnschedulable {
+
+			isUnschedulable = true
+			break
+		}
+	}
+
+	if !isUnschedulable {
+		return nil
+	}
+
+	now := time.Now().UTC().Unix()
+	creationTime := pod.CreationTimestamp.Time.UTC().Unix()
+	diffSeconds := now - creationTime
+	if diffSeconds < 0 {
+		diffSeconds = 0
+	}
+
+	migrationKey, err := controller.KeyFunc(migration)
+	if err != nil {
+		return err
+	}
+
+	if diffSeconds >= c.unschedulableTimeoutSeconds {
+		c.podExpectations.ExpectDeletions(migrationKey, []string{controller.PodKey(pod)})
+		err = c.clientset.CoreV1().Pods(vmi.Namespace).Delete(context.Background(), pod.Name, v1.DeleteOptions{})
+		if err != nil {
+			c.podExpectations.DeletionObserved(migrationKey, controller.PodKey(pod))
+			c.recorder.Eventf(migration, k8sv1.EventTypeWarning, FailedDeletePodReason, "Error deleted migration target pod: %v", err)
+			return fmt.Errorf("failed to delete vmi migration target pod stuck in unschedulable state: %v", err)
+		}
+		log.Log.Object(vmi).Infof("Deleted pending unschedulable migration target pod %s/%s with uuid %s for migration %s with uuid %s", pod.Namespace, pod.Name, string(pod.UID), migration.Name, string(migration.UID))
+		c.recorder.Eventf(migration, k8sv1.EventTypeNormal, SuccessfulDeletePodReason, "Deleted pending migration target pod %s due to being unschedulable.", pod.Name)
+	} else {
+		// Make sure we check this again after some time
+		c.Queue.AddAfter(migrationKey, time.Second*time.Duration(c.unschedulableTimeoutSeconds-diffSeconds))
```

Comment:
Maybe I'm just too thick, but I don't follow this `c.unschedulableTimeoutSeconds-diffSeconds` . 

Would you assign it to a named variable to help people like me reason this out ?

Reply:
i changed this around a bit to try and help

Reply:
Tks. Better now.